packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarGroupOwnProps
+>;
+
+const ToolbarGroup = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="group"
+      selector={getSelector(GROUP_NAME)}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarGroupPrimitive;
+
+ToolbarGroup.displayName = GROUP_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarLabel
+ * -----------------------------------------------------------------------------------------------*/
+
+const LABEL_NAME = 'ToolbarLabel';
+
+type ToolbarLabelOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarLabelPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarLabelOwnProps
+>;
+
+const ToolbarLabel = React.forwardRef((props, forwardedRef) => (
+  <Primitive selector={getSelector(LABEL_NAME)} {...props} ref={forwardedRef} />
+)) as ToolbarLabelPrimitive;
+
+ToolbarLabel.displayName = LABEL_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarSeparator
+ * -----------------------------------------------------------------------------------------------*/
+
+const SEPARATOR_NAME = 'ToolbarSeparator';
+
+type ToolbarSeparatorOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarSeparatorPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarSeparatorOwnProps
+>;
+
+const ToolbarSeparator = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="separator"
+      selector={getSelector(SEPARATOR_NAME)}
+      aria-orientation={orientation}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarSeparatorPrimitive;
+
+ToolbarSeparator.displayName = SEPARATOR_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarItem
+ * -----------------------------------------------------------------------------------------------*/
+
+const ITEM_NAME = 'ToolbarItem';
+
+type ToolbarItemOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarItemOwnProps
+>;
+
+const ToolbarItem = React.forwardRef((props, forwardedRef) => {
+  const { disabled, ...itemProps } = props as any;
+
+  const rovingFocusProps = useRovingFocus({
+    disabled: disabled,
+    active: false,
+  });
+
+  const handleClick = (event: MouseEvent) => {
+    if (disabled) {
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <Primitive
+      ref={forwardedRef}
```

Comment:
we tend to pay attention to the order in the spread here, ie. we let users override certain props but not others.

As a rule of thumb, we let them override `aria-` stuff, `role`, our `data-` attributes and then we spread the props, then straight after we'd have first `selector` then `ref`, then the rest.

Reply:
I would have never thought of that, thanks!